envoy/listener.go

```diff
@@ -0,0 +1,45 @@
+package envoy
+
+import (
+	"github.com/Nitro/sidecar/catalog"
+)
+
+const (
+	listenerEventBufferSize = 100
+)
+
+// Listener is an internal Sidecar listener that will be hooked up when
+// config.Envoy.UseGRPCAPI is true. It only needs to know when the state
+// changes, ignoring what the actual change was.
+type Listener struct {
+	eventsChan chan catalog.ChangeEvent
+}
+
+// Chan exposes the internal events channel
+func (l *Listener) Chan() chan catalog.ChangeEvent {
+	return l.eventsChan
+}
+
+// Name returns a unique name for this listener
+func (l *Listener) Name() string {
+	// Be careful not to clash with names assigned automatically in
+	// service/service.go -> ListenerName()
+	return "internal-envoy"
+}
+
+// Managed tells Sidecar that it shouldn't try to automatically remove
+// this listener
+func (l *Listener) Managed() bool {
+	return false
+}
+
```

Comment:
Let's go with this implementation since you built it and it works! I was just thinking that if we have any sync issues, we could run this on a timed loop instead of on listener changes. That would remove the issue of de-duping, and is actually how Envoy works (it pulls us).